nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/UpdateDatabaseTable.java

```diff
@@ -0,0 +1,606 @@
+            .name("updatedatabasetable-dbcp-service")
+            .displayName("Database Connection Pooling Service")
+            .description("The Controller Service that is used to obtain connection(s) to the database")
+            .required(true)
+            .identifiesControllerService(DBCPService.class)
+            .build();
+
+    static final PropertyDescriptor CATALOG_NAME = new PropertyDescriptor.Builder()
+            .name("put-db-record-catalog-name")
+            .displayName("Catalog Name")
+            .description("The name of the catalog that the statement should update. This may not apply for the database that you are updating. In this case, leave the field empty. Note that if the "
+                    + "property is set and the database is case-sensitive, the catalog name must match the database's catalog name exactly.")
+            .required(false)
+            .expressionLanguageSupported(FLOWFILE_ATTRIBUTES)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    static final PropertyDescriptor SCHEMA_NAME = new PropertyDescriptor.Builder()
+            .name("put-db-record-schema-name")
+            .displayName("Schema Name")
+            .description("The name of the database schema that the table belongs to. This may not apply for the database that you are updating. In this case, leave the field empty. Note that if the "
+                    + "property is set and the database is case-sensitive, the schema name must match the database's schema name exactly.")
+            .required(false)
+            .expressionLanguageSupported(FLOWFILE_ATTRIBUTES)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    static final PropertyDescriptor TABLE_NAME = new PropertyDescriptor.Builder()
+            .name("updatedatabasetable-table-name")
+            .displayName("Table Name")
+            .description("The name of the database table to update. If the table does not exist, then it will either be created or an error thrown, depending "
+                    + "on the value of the Create Table property.")
+            .required(true)
+            .expressionLanguageSupported(ExpressionLanguageScope.FLOWFILE_ATTRIBUTES)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    static final PropertyDescriptor CREATE_TABLE = new PropertyDescriptor.Builder()
+            .name("updatedatabasetable-create-table")
+            .displayName("Create Table Strategy")
+            .description("Specifies how to process the target table when it does not exist (create it, fail, e.g.).")
+            .required(true)
+            .addValidator(Validator.VALID)
+            .allowableValues(CREATE_IF_NOT_EXISTS, FAIL_IF_NOT_EXISTS)
+            .defaultValue(FAIL_IF_NOT_EXISTS.getValue())
+            .build();
+
+    static final PropertyDescriptor TRANSLATE_FIELD_NAMES = new PropertyDescriptor.Builder()
```

Comment:
Translate and Update are similar but are used for different things. Translate is to normalize the field names so they can be matched against the (normalized) column names to see if there's an existing column. Update Field Names means if there's an existing column that matches (whether Translate is true or false), update the record field names to ensure they match the exact column names. I'm not sure this is required for all databases (I know it is for Hive) but if we leave it in then the downstream PutDatabaseTable wouldn't need to set Translate Field Names to true because they'd match already. I agree it's a lot less important than it was for UpdateHive3Table but I figured it was better to be more flexible/robust just in case.